Failure delivering result ResultInfo{who=null, request=65538, result=0, data=null} to activity {fr.free.nrw.commons/fr.free.nrw.commons.contributions.ContributionsActivity}: java.lang.NullPointerException

By Антон Обожин on 20th July:

```
USER_COMMENT=I pressed photo icon at top right, then back button and app crashed
ANDROID_VERSION=4.2.2
APP_VERSION_NAME=1.19
BRAND=Lenovo
PHONE_MODEL=Lenovo A859
CUSTOM_DATA=
STACK_TRACE=java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=65538, result=0, data=null} to activity {fr.free.nrw.commons/fr.free.nrw.commons.contributions.ContributionsActivity}: java.lang.NullPointerException
at android.app.ActivityThread.deliverResults(ActivityThread.java:3566)
at android.app.ActivityThread.handleSendResult(ActivityThread.java:3609)
at android.app.ActivityThread.access$1100(ActivityThread.java:169)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1425)
at android.os.Handler.dispatchMessage(Handler.java:107)
at android.os.Looper.loop(Looper.java:194)
at android.app.ActivityThread.main(ActivityThread.java:5449)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:525)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:833)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:600)
at dalvik.system.NativeStart.main(Native Method)
Caused by: java.lang.NullPointerException
at fr.free.nrw.commons.contributions.ContributionsListFragment.onActivityResult(ContributionsListFragment.java:95)
at android.support.v4.app.FragmentActivity.onActivityResult(FragmentActivity.java:175)
at android.app.Activity.dispatchActivityResult(Activity.java:5324)
at android.app.ActivityThread.deliverResults(ActivityThread.java:3562)
... 11 more
java.lang.NullPointerException
at fr.free.nrw.commons.contributions.ContributionsListFragment.onActivityResult(ContributionsListFragment.java:95)
at android.support.v4.app.FragmentActivity.onActivityResult(FragmentActivity.java:175)
at android.app.Activity.dispatchActivityResult(Activity.java:5324)
at android.app.ActivityThread.deliverResults(ActivityThread.java:3562)
at android.app.ActivityThread.handleSendResult(ActivityThread.java:3609)
at android.app.ActivityThread.access$1100(ActivityThread.java:169)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1425)
at android.os.Handler.dispatchMessage(Handler.java:107)
at android.os.Looper.loop(Looper.java:194)
at android.app.ActivityThread.main(ActivityThread.java:5449)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:525)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:833)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:600)
at dalvik.system.NativeStart.main(Native Method)
```

By Don Hamza on 20th July:

```
USER_COMMENT=تصوير شمس
ANDROID_VERSION=4.4.2
APP_VERSION_NAME=1.19
BRAND=Lenovo
PHONE_MODEL=Lenovo A319
CUSTOM_DATA=
STACK_TRACE=java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=65538, result=-1, data=null} to activity {fr.free.nrw.commons/fr.free.nrw.commons.contributions.ContributionsActivity}: java.lang.NullPointerException
at android.app.ActivityThread.deliverResults(ActivityThread.java:3553)
at android.app.ActivityThread.handleSendResult(ActivityThread.java:3596)
at android.app.ActivityThread.access$1300(ActivityThread.java:151)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1369)
at android.os.Handler.dispatchMessage(Handler.java:110)
at android.os.Looper.loop(Looper.java:193)
at android.app.ActivityThread.main(ActivityThread.java:5299)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:515)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:829)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:645)
at dalvik.system.NativeStart.main(Native Method)
Caused by: java.lang.NullPointerException
at fr.free.nrw.commons.contributions.ContributionsListFragment.onActivityResult(ContributionsListFragment.java:95)
at android.support.v4.app.FragmentActivity.onActivityResult(FragmentActivity.java:175)
at android.app.Activity.dispatchActivityResult(Activity.java:5456)
at android.app.ActivityThread.deliverResults(ActivityThread.java:3549)
... 11 more
java.lang.NullPointerException
at fr.free.nrw.commons.contributions.ContributionsListFragment.onActivityResult(ContributionsListFragment.java:95)
at android.support.v4.app.FragmentActivity.onActivityResult(FragmentActivity.java:175)
at android.app.Activity.dispatchActivityResult(Activity.java:5456)
at android.app.ActivityThread.deliverResults(ActivityThread.java:3549)
at android.app.ActivityThread.handleSendResult(ActivityThread.java:3596)
at android.app.ActivityThread.access$1300(ActivityThread.java:151)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1369)
at android.os.Handler.dispatchMessage(Handler.java:110)
at android.os.Looper.loop(Looper.java:193)
at android.app.ActivityThread.main(ActivityThread.java:5299)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:515)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:829)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:645)
at dalvik.system.NativeStart.main(Native Method)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,8 @@
 # Wikimedia Commons for Android
 
+##v1.20
+- Hotfix for data=null crash
+
 ##v1.19
 - Fixed adapter crash
 - Attempt at fixing Google Photos crash

--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -1,7 +1,7 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="fr.free.nrw.commons"
-    android:versionCode="37"
-    android:versionName="1.19" >
+    android:versionCode="38"
+    android:versionName="1.20" >
 
     <uses-permission android:name="android.permission.INTERNET" />
     <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>

--- a/app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListFragment.java
+++ b/app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListFragment.java
@@ -92,8 +92,10 @@ public class ContributionsListFragment extends Fragment {
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
         //FIXME: must get the file data for Google Photos when receive the intent answer, in the onActivityResult method
         super.onActivityResult(requestCode, resultCode, data);
-        Uri imageData = data.getData();
-        controller.handleImagePicked(requestCode, imageData);
+        if (data != null) {
+            Uri imageData = data.getData();
+            controller.handleImagePicked(requestCode, imageData);
+        }
     }
 
 

--- a/app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListFragment.java
+++ b/app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListFragment.java
@@ -94,11 +94,11 @@ public class ContributionsListFragment extends Fragment {
         super.onActivityResult(requestCode, resultCode, data);
 
         if (data != null) {
-            Log.d("Contributions", "OnActivityResult() parameters: Result code: " + resultCode + "Data: " + data.toString());
+            Log.d("Contributions", "OnActivityResult() parameters: Result code: " + resultCode + " Data: " + data.toString());
             Uri imageData = data.getData();
             controller.handleImagePicked(requestCode, imageData);
         } else {
-            Log.d("Contributions", "OnActivityResult() parameters: Result code: " + resultCode + "Data: null");
+            Log.d("Contributions", "OnActivityResult() parameters: Result code: " + resultCode + " Data: null");
         }
     }
 

--- a/app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListFragment.java
+++ b/app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListFragment.java
@@ -92,9 +92,13 @@ public class ContributionsListFragment extends Fragment {
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
         //FIXME: must get the file data for Google Photos when receive the intent answer, in the onActivityResult method
         super.onActivityResult(requestCode, resultCode, data);
+
         if (data != null) {
+            Log.d("Contributions", "OnActivityResult() parameters: Result code: " + resultCode + "Data: " + data.toString());
             Uri imageData = data.getData();
             controller.handleImagePicked(requestCode, imageData);
+        } else {
+            Log.d("Contributions", "OnActivityResult() parameters: Result code: " + resultCode + "Data: null");
         }
     }
 

